internal/controller/object/indexes.go

```diff
@@ -123,8 +123,10 @@ func enqueueObjectsForReferences(ca cache.Cache, log logging.Logger) func(ctx co
 		}
 		// queue those Objects for reconciliation
 		for _, o := range objects.Items {
-			log.Info("Enqueueing Object because referenced resource changed", "name", o.GetName(), "referencedGVK", rGVK.String(), "referencedName", ev.Object.GetName(), "providerConfig", pc)
-			q.Add(reconcile.Request{NamespacedName: types.NamespacedName{Name: o.GetName()}})
+			if o.Spec.Watch {
```

Comment:
why is this necessary?

Reply:
We only want to enqueue the Object if `spec.watch` is set to true. Not every Object wants to watch the referenced resources. If you're questioning why we have such a field in the first place, I have a comment for that [here](https://github.com/crossplane-contrib/provider-kubernetes/pull/235#discussion_r1589157066).

Reply:
Not questioning the watch field. But why we make the `q.Add` conditional on it. I understand that we don't start watches on their referenced resources, but here we are not doing that, but queuing them.

Reply:
Ah, I see. 

Watches started based on GVK not including their namespaced name, so, another object with `spec.watch: true` might have started a watch on that given GVK. For example, think two Objects; one referencing `secret: foo` with `watch: true`, another one referencing `secret: bar` with `watch: false`. Even though the latter does not start watch on Secrets, it was started with the former. So, we need to filter out here, while adding to the queue otherwise the latter object will be getting watch events even not desired.